src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -102,12 +103,64 @@ const registerAndPersistInvoice = async ({
   return invoice
 }
 
-const walletIdFromWalletName = async (
-  walletName: WalletName,
+const walletIdFromPublicId = async (
+  walletPublicId: WalletPublicId,
+): Promise<WalletId | RepositoryError> => {
+  const walletsRepo = WalletsRepository()
+  const wallet = await walletsRepo.findByPublicId(walletPublicId)
+  if (wallet instanceof Error) return wallet
+
+  return wallet.id
+}
+
+// TODO: Remove. Used in v1 only
+const walletIdFromUsername = async (
+  username: Username,
 ): Promise<WalletId | RepositoryError> => {
   const walletsRepo = WalletsRepository()
-  const wallet = await walletsRepo.findByWalletName(walletName)
+  const wallet = await walletsRepo.findByUsername(username)
   if (wallet instanceof Error) return wallet
 
   return wallet.id
 }
+
+// TODO: Remove. Used in v1 only
+export const addInvoiceForUsername = async ({
```

Comment:
Making a note here about the tradeoffs of this approach (I'm assuming it's simplicity) of going straight from `username` to `walletId` in this separate `addInvoiceForUsername` method vs. having a `username` resolver that can get us a `walletPublicId` and then using that resolved `walletPublicId` to call the `addInvoiceForRecipient` method.

I'm thinking of this within the context of if we decide to go with the `handle` approach we were talking about. I'm ok with this change here for now though since it's only a v1 thing.

Reply:
I moved them under core